src/device-state/current-state.ts

```diff
@@ -138,128 +135,109 @@ const getStateDiff = (): DeviceStatus => {
-				: {
-						cpu_usage: null,
-						memory_usage: null,
-						memory_total: null,
-						storage_usage: null,
-						storage_total: null,
-						storage_block_device: null,
-						cpu_temp: null,
-						cpu_id: null,
-				  }),
-			...(await sysInfo.getSystemChecks()),
-		};
+async function generateStateForReport() {
+	const { hardwareMetrics } = await config.getMany(['hardwareMetrics']);
+
+	const currentDeviceState = await deviceState.getStatus();
+
+	// If hardwareMetrics is false, send null patch for system metrics to cloud API
+	const info = {
+		...(hardwareMetrics
+			? await sysInfo.getSystemMetrics()
+			: {
+					cpu_usage: null,
+					memory_usage: null,
+					memory_total: null,
+					storage_usage: null,
+					storage_total: null,
+					storage_block_device: null,
+					cpu_temp: null,
+					cpu_id: null,
+			  }),
+		...(await sysInfo.getSystemChecks()),
+	};
 
-		stateForReport.local = {
-			...stateForReport.local,
+	return {
+		local: {
 			...currentDeviceState.local,
 			...info,
-		};
-		stateForReport.dependent = {
-			...stateForReport.dependent,
-			...currentDeviceState.dependent,
-		};
-
-		const stateDiff = getStateDiff();
-
-		// report state diff
-		throttledReport(
-			{ stateDiff, conf },
-			() => {
-				// If the patch succeeds update lastReport and reset the error counter
-				_.assign(lastReportedState.local, stateDiff.local);
-				_.assign(lastReportedState.dependent, stateDiff.dependent);
-				stateReportErrors = 0;
-
-				reportPending = false;
-			},
-			(e) => {
-				if (e instanceof StatusError) {
-					// We don't want these errors to be classed as a report error, as this will cause
-					// the watchdog to kill the supervisor - and killing the supervisor will
-					// not help in this situation
-					log.error(
-						`Device state report failure! Status code: ${e.statusCode} - message:`,
-						e.message,
-					);
-
-					// We want to backoff on all errors, but without having the healthchecks
-					// get triggered.
-					// This will use retryAfter as maxDelay if the header is present in the
-					// response by the API
-					backoff(
-						reportCurrentState,
-						// Do not do exponential backoff if the API reported a retryAfter
-						e.retryAfter ? 0 : attempts,
-						maxDelay,
-						// Start the polling at the value given by the API if any
-						e.retryAfter ?? MINIMUM_BACKOFF_DELAY,
-					);
-				} else {
-					eventTracker.track('Device state report failure', {
-						error: e.message,
-					});
-					backoff(reportCurrentState, stateReportErrors++, maxDelay);
-				}
-			},
-		);
-	})();
+		},
+		dependant: currentDeviceState.dependent,
+	};
 }
 
-export const startReporting = () => {
-	const doReport = () => {
+export async function startReporting() {
+	// Get configs needed to make a report
+	const reportConfigs = (await config.getMany([
+		'uuid',
+		'apiEndpoint',
+		'apiTimeout',
+		'deviceApiKey',
+		'deviceId',
+		'localMode',
+		'appUpdatePollInterval',
+		'hardwareMetrics',
```

Comment:
don't need this in report config since it's only used for getting the target state...but maybe we want to pass this config in so getting target state doesn't have to query the config...?

Reply:
leaving this for now since I don't want to add more files being modified to this PR